2D Components

daredevildave opened the pull request:
Not a bad place to review this stuff before I finish up the final bits and pieces. This PR contains 4 new components, screen, element, image and text.

Code things you might want to review:
- Added msdf and screenspace rendering to StandardMaterial
- element component overrides entity.sync to generate correct transforms
- explicit drawcall ordering done for elements using meshInstance.drawOrder
- Still haven't decided on the final name for the "screen" component

Usage-wise, I suggest you checkout the branch and look at /examples/ui/ which is mad test case for all the different anchors. I'll tidy that example up before release.

guycalledfrank commented on src/scene/forward-renderer.js:
```diff
@@ -1122,9 +1124,13 @@ pc.extend(pc, function () {
+
                         drawCalls[j] = prevDrawCall;
                         drawCalls[j - 1] = drawCall;
-                        j--;
```
Hmm not sure why you moved the decrement

daredevildave replied:
Good catch. It was to fix an infinite loop. I'll update it.

daredevildave commented:
Updated again today

- moved text and image components into element component
- loads of other fixes and additions including proper support for anchors

I'm going to merge this as a "hidden beta" later on today unless anyone objects.
